winter_openapi/generator.py

```diff
@@ -51,23 +51,30 @@ def generate_openapi(
     tag_names = [tag_.name for tag_ in tags_]
     paths: Paths = {}
     operation_ids: Set[str] = set()
-    for key, group_routes in groupby(routes, key=lambda r: r.url_path):
-        path = _get_openapi_path(route=group_routes, operation_ids=operation_ids, tag_names=tag_names)
-        paths[key] = path
+    path_prefix = determine_path_prefix(routes)
+
+    for url_path, group_routes in groupby(routes, key=lambda r: r.url_path):
+        url_path_without_prefix = get_url_path_without_prefix(url_path, path_prefix)
+        path_item = _get_openapi_path(routes=group_routes, operation_ids=operation_ids, tag_names=tag_names, path_prefix=path_prefix)
+        paths[url_path_without_prefix] = path_item
 
     info = Info(title=title, version=version, description=description)
     servers_ = [Server(**server) for server in servers or []]
     open_api = OpenAPI(info=info, servers=servers_, paths=paths, components=components, tags=tags_)
     return open_api.dict(by_alias=True, exclude_none=True)
 
 
-def _get_openapi_path(*, route: Iterable[Route], operation_ids: Set[str], tag_names: Iterable[str]) -> PathItem:
+def _get_openapi_path(*, routes: Iterable[Route], operation_ids: Set[str], tag_names: Iterable[str], path_prefix: str) -> PathItem:
     path = {}
-    for route_item in route:
+    for route_item in routes:
         operation_id = route_item.method.full_name
         if operation_id in operation_ids:
             warnings.warn(f"Duplicate Operation ID {operation_id}")
         operation_ids.add(operation_id)
+        url_path_tag = get_url_path_tag(route_item, path_prefix)
```

Comment:
Maybe pass already calculated tag to _get_openapi_path method? instead of the path_prefix

Reply:
Do you mean this because it is a group of routes?

Reply:
Yes, group of routes with the same url. Also it will remove the argument and make the code simpler.